src/layers/scenegraph-layer/scenegraph-layer.js

```diff
@@ -0,0 +1,312 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {ScenegraphLayer as DeckScenegraphLayer} from '@deck.gl/mesh-layers';
+import {load} from '@loaders.gl/core';
+import {GLTFScenegraphLoader} from '@luma.gl/addons';
+
+import Layer from '../base-layer';
+import memoize from 'lodash.memoize';
+import ScenegraphLayerIcon from './scenegraph-layer-icon';
+
+export const scenegraphRequiredColumns = ['lat', 'lng'];
+export const scenegraphOptionalColumns = ['altitude'];
+
+export const scenegraphPosAccessor = ({lat, lng, altitude}) => d => [
+  // lng
+  d.data[lng.fieldIdx],
+  // lat
+  d.data[lat.fieldIdx],
+  // altitude
+  altitude && altitude.fieldIdx > -1 ? d.data[altitude.fieldIdx] : 0
+];
+
+export const scenegraphPosResolver = ({lat, lng, altitude}) =>
+  `${lat.fieldIdx}-${lng.fieldIdx}-${altitude ? altitude.fieldIdx : 'z'}`;
+
+export const scenegraphVisConfigs = {
+  radius: 'radius',
+  fixedRadius: 'fixedRadius',
+  opacity: 'opacity',
+  colorRange: 'colorRange',
+  radiusRange: 'radiusRange',
+  //
+  sizeScale: 'sizeScale',
+  angleX: 'angleX',
+  angleY: 'angleY',
+  angleZ: 'angleZ'
+};
+
+export default class ScenegraphLayer extends Layer {
+  constructor(props) {
+    super(props);
+
+    this.registerVisConfig(scenegraphVisConfigs);
+    this.getPosition = memoize(scenegraphPosAccessor, scenegraphPosResolver);
+
+    // prepare layer info modal
+    // this._layerInfoModal = IconInfoModalFactory();
+    // this.getSvgIcons();
+  }
+
+  get type() {
+    return '3D';
+  }
+
+  get requiredLayerColumns() {
+    return scenegraphRequiredColumns;
+  }
+
+  get optionalColumns() {
+    return scenegraphOptionalColumns;
+  }
+
+  get columnPairs() {
+    return this.defaultPointColumnPairs;
+  }
+
+  get layerIcon() {
+    return ScenegraphLayerIcon;
+  }
+
+  get visualChannels() {
+    return {
+      ...super.visualChannels,
+      size: {
+        ...super.visualChannels.size,
+        range: 'radiusRange',
+        property: 'radius',
+        channelScaleType: 'radius'
+      }
+    };
+  }
+
+  get layerInfoModal() {
```

Comment:
If you don't plan to add any modal to explain how to draw a screen graph layer, you don't need this. But I think  we can create a simple modal that explains the supported modal format and put a screenshot there :)

This is what icon layer has:
<img width="318" alt="Screen Shot 2019-06-03 at 1 54 19 PM" src="https://user-images.githubusercontent.com/3605556/58834016-22e77d00-8607-11e9-8341-dd92087d011d.png">

<img width="1444" alt="Screen Shot 2019-06-03 at 1 54 27 PM" src="https://user-images.githubusercontent.com/3605556/58834030-2bd84e80-8607-11e9-8565-8f399504d585.png">

Reply:
<img width="913" alt="Screen Shot 2019-06-07 at 3 47 11 PM" src="https://user-images.githubusercontent.com/25012840/59137222-88a37400-893b-11e9-8c49-f9cf7130e970.png">